pwndbg/commands/version.py

```diff
@@ -259,3 +185,67 @@ def bugreport(run_browser=False, use_gh=False):
             print(please_please_submit + github_issue_url)
     else:
         print(please_please_submit + github_issue_url)
+
+
+def get_debugger_configuration():
+    if pwndbg.dbg.is_gdblib_available():
+        import gdb
+        gdb_config = gdb.execute("show configuration", to_string=True).split("\n")
+        return "\n" + "\n".join(gdb_config)
+
+    # LLDB: TODO/FIXME: Do we need this?
+    else:
+        return ""
+
+
+def get_debugger_session_history():
+    if pwndbg.dbg.is_gdblib_available():
+        import gdb
+
+        # get saved history size (not including current gdb session)
+        gdb_history_file = gdb.execute("show history filename", to_string=True)
+        gdb_history_file = gdb_history_file[
+            gdb_history_file.index('"') + 1 : gdb_history_file.rindex('"')
+        ]
+        gdb_history_len = 0
+        try:
+            with open(gdb_history_file) as f:
+                gdb_history_len = len(f.readlines())
+        except FileNotFoundError:
+            pass
+
+        max_command_no = 0
+        history_commands = gdb.execute("show commands", to_string=True)
+        if history_commands:
+            history_commands = history_commands.split("\n")
+            if len(history_commands) > 1:
+                # The last element of the list is the `show commands` command we
+                # just ran, so we need to get the second to last one
+                last_command = history_commands[-2]
+                max_command_no = int(last_command.split()[0]) - 1
+
+        show_command_size = 10  # 'show command' returns 10 commands
+        gdb_current_session_history = {}
+        current_command_no = gdb_history_len + 1
+
+        while current_command_no <= max_command_no:
+            cmds = gdb.execute(
+                "show commands " + str(current_command_no + (show_command_size // 2) + 1),
+                to_string=True,
+            ).split("\n")[:-1]
+            for cmd in cmds:
+                cmd_no, cmd = cmd.split(maxsplit=1)
+                cmd_no = int(cmd_no)
+                if cmd_no <= gdb_history_len:
+                    continue
+                if current_command_no > max_command_no:
+                    break
+                gdb_current_session_history[cmd_no] = cmd
+                current_command_no += 1
+
+        gdb_current_session_history = (v for (k, v) in sorted(gdb_current_session_history.items()))
+        return "\n".join(gdb_current_session_history)
+
+    # LLDB: TODO/FIXME: Not yet supported
+    else:
+        return "<session history not supported on lldb yet>"
```

Comment:
maybe `readline.get_current_history_length()`

Reply:
wow, wondering if that'd work for gdb too since what we do is crazy :p

Reply:
readline module in gdb is banned